Switch DNS addons from skydns to kubedns

girishkalele opened the pull request:
Change GCI and trusty cluster-helper scripts to use kubedns instead of skydns.

luxas commented on cluster/addons/dns/kubedns-rc.yaml.in:
```diff
@@ -0,0 +1,80 @@
+          name: dns-tcp
+          protocol: TCP
+      - name: healthz
+        image: gcr.io/google_containers/exechealthz:1.0
```
`exechealthz-amd64:1.0`

luxas commented on cluster/gce/coreos/kube-manifests/addons/dns/kubedns-rc.yaml:
```diff
@@ -0,0 +1,80 @@
+          name: dns-tcp
+          protocol: TCP
+      - name: healthz
+        image: gcr.io/google_containers/exechealthz:1.0
```
`exechealthz-amd64:1.0`

bprashanth commented:
Can you update the RELEASE guide with how to build a new image (there's one in the kube2sky subdir, not sure how we want to documenting releasing this though, maybe add it to build/kube-dns?), and add a comment in skydns.rc that it's deprecated. I'd also prefer just having one kubedns.rs, so maybe remove the deployment in cluster/saltbase for now?

roberthbailey commented:
lgtm once you've addressed comments from @luxas

girishkalele commented:
Noticed a new merge to master - some more changes to the kubedns-rc saltbase files.

@ArtfulCoder - after the switch to GCI images, the new kubedns pod is running anymore when I do kube-up. Its running the old skydns pod. I am trying to fix the files in the addons directory and cluster-helper to switch to kube-dns.

ArtfulCoder commented:
1. lets pick 1.2 kubedns image
2. lets have the dnsmasq container too.
https://github.com/ArtfulCoder/kubernetes/blob/a7bf2455782695924cdee50f6818e18da788b5f5/cluster/saltbase/salt/kube-dns/kubedns-rc.yaml.in

thockin commented on cluster/addons/dns/kubedns-rc.yaml.in:
```diff
@@ -0,0 +1,80 @@
+apiVersion: v1
+kind: ReplicationController
```
isn't this basically the same as cluster/saltbase/salt/kube-dns/kubedns-rc.yaml.in ?  Why do we need both?  Can we just kill off cluster/addons/dns and leave a pointer to the new location?

thockin replied:
also see  #25020

thockin commented:
I'm all for expedient solutions, but this seems  redundant, doesn't it?  can we discuss quickly today?

girishkalele commented:
@thockin 
Yes, this was getting more confusing with all the duplication.
Yesterday I took a step back to take a look at all the duplication of the skydns-rc.yaml.in and skydns-rc.yaml and understand how these are getting packaged into the manifests tarball for GCE/GKE vs other platforms.
I believe the the old directories cluster/addons/dns/(kube2sky|skydns) also need to go away since they are no longer in use.

Commit:
Switch DNS addons from skydns to kubedns

Unified skydns templates using a simple underscore based template and
added transform sed scripts to transform into salt and sed yaml
templates

Moved all content out of cluster/addons/dns into build/kube-dns and
saltbase/salt/kube-dns

girishkalele commented:
Was labelled XXL because of the large deletions of deprecated kube2sky and skydns directories.

bprashanth commented on cluster/addons/dns/skydns-rc.yaml.in:
```diff
@@ -1,62 +1,49 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
```
if you're doing this anyway why not just get rid of addons and move to cluster/saltbase instead? (IMO addons/ for DNS is confusing, and I thought previously you wanted to keep the change small)

girishkalele replied:
Tim suggested the same (remove from 'addons')

I am deleting the cluster/addons/dns directory completely and moving templates to saltbase and build code to build/kube-dns.

bprashanth commented on docs/devel/running-locally.md:
````diff
@@ -189,7 +189,7 @@ KUBE_DNS_DOMAIN="cluster.local"
 KUBE_DNS_REPLICAS=1
 ```
 
-To know more on DNS service you can look [here](http://issue.k8s.io/6667). Related documents can be found [here](../../cluster/addons/dns/#how-do-i-configure-it)
````
can you also update local-up-cluster?

bprashanth commented on cluster/saltbase/salt/kube-dns/skydns-svc.yaml.sed:
```diff
@@ -12,8 +12,27 @@
-ADD kube2sky /
-ADD kube2sky.go /
-ENTRYPOINT ["/kube2sky"]
+# This file should be kept in sync with cluster/addons/dns/skydns-rc.yaml.in
```
skydns is a different container https://hub.docker.com/r/skynetservices/skydns/, I think we should just call this kubedns-rc/svc etc.

girishkalele replied:
This is a messed-up git rename detection....it thinks the Dockerfile is renamed to a yaml template !

cluster/addons/dns/kube2sky/Dockerfile → ...altbase/salt/kube-dns/skydns-svc.yaml.sed

bprashanth commented on cluster/saltbase/salt/kube-dns/skydns-svc.yaml.in:
```diff
@@ -0,0 +1,38 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This file should be kept in sync with cluster/addons/dns/skydns-rc.yaml.in
```
isn't this false?

bprashanth commented on cluster/addons/dns/kube2sky/RELEASES.md:
```diff
@@ -1,43 +0,0 @@
-release, here are the steps for making a release. We make releases when they
-are ready, not on every PR.
-
-1. Build the container for testing: `make container PREFIX=<your-docker-hub> TAG=rc`
```
I think these release steps are still valid for kubedns, we just need to modify to say something like: first build the binary with ./hack/build-go, then cd build/kube-dns and run make etc.

bprashanth commented on cluster/addons/dns/kube2sky/Changelog:
```diff
@@ -1,40 +0,0 @@
-## Version 1.15 (Apr 7 2016 Lucas Käldström <lucas.kaldstrom@hotmail.co.uk>)
```
A similar changelog for build/kube-dns would be nice

girishkalele commented:
@bprashanth 

Changing the filename from skydns-rc/svc to kubedns-rc/svc rename will be a much noisier change, I held off on renaming it for that reason.

Commit:
Address review comments

bprashanth commented:
> Changing the filename from skydns-rc/svc to kubedns-rc/svc rename will be a much noisier change, I held off on renaming it for that reason.

If it's not just a simple rename, I'm fine with a follow up, as long as you add a todo somewhere. I don't want people to see skydns as a pod name when they do kubectl get pods because that's misleading. 

I think we should get this in asap so we can start testing it asap, hopefully before we cut the branch.

bprashanth commented on cluster/saltbase/salt/kube-dns/skydns-rc.yaml.in:
```diff
@@ -53,7 +71,7 @@ spec:
         args:
         # command = "/kube-dns"
-        - --domain={{ pillar['dns_domain'] }}.
+        - --domain={{ pillar['dns_domain'] }}
```
are you removing the "." ?

girishkalele replied:
Good catch...the . needs to be there.

bprashanth commented on cluster/images/hyperkube/addons/dns-rc.yaml:
```diff
@@ -12,66 +12,39 @@
 kind: ReplicationController
 metadata:
-  name: kube-dns-v11
+  name: kube-dns-v13
```
what happened to v12?

girishkalele replied:
Keeping hyperkube in sync with the skydns-rc bump to v13.

bprashanth commented:
This is your issue if you need a unittest rerun and are not going to upload a new pr (https://github.com/kubernetes/kubernetes/issues/26578)

girishkalele commented:
Oh thanks...I didn't see the test failure - it seems like it was a flake, I will submit the review changes and rerun the tests.

Commit:
Code review comments and corrections

bprashanth commented:
We should fix the v13 x image: 1.2 offset, but not necessarily in this pr since there will be more pushes. I also guess this doesn't need a release note since we're adding one for the inital kube-dns.

bprashanth commented:
Marking p2 since DNS is currently getting no test coverage
